### PR TITLE
Trainer slide ins

### PR DESCRIPTION
**Pull request by assassinsgreed**
Trainer slide-in messages for all known trainers right now, with placeholders for an unknown number of rival battles.
No messages are linked to trainer IDs yet.

Also includes base config changes for:
- Syncing # of TMS/HMs/Tutors with DPE
- Syncing moves list with DPE

### CODE REVIEW

File: strings/trainer_sliding_strings.string [hunk 1]
```diff
@@ -1,41 +1,202 @@
...
+// Rival 1
+#org @sText_Rival1_LastSwitchIn
+Okay, [FD][06]! Let's do this!\p 
+
+#org @sText_Rival1_LowHP
+No way! [FD][06] and I\naren't done yet!\p 
+
+// Rival n here!
+
+// Leader 1
```
Comment (CrystalAQuick) [inline strings/trainer_sliding_strings.string:24]: could make the comment like
// Gym Leader 1 (Terrence) (here as well as other places for the other leaders)

File: strings/trainer_sliding_strings.string [hunk 2]
```diff
@@ -1,41 +1,202 @@
...
+#org @sText_Kurtis_FirstMonDown
+I can see why you defeated Irene\nand Roland. You'd best not\nunderestimate me, though.\p 
+
+#org @sText_Kurtis_LastSwitchIn
+Team Pluto won't fall to the likes\nof you. Not now, not ever.\p 
+
+#org @sText_Kurtis_LowHP
+Now you're making me angry!\p 
+
+// Alistair
```
Comment (CrystalAQuick) [inline strings/trainer_sliding_strings.string:134]: use the same comment for alistair
// Pokemon Trainer Alistair

File: strings/trainer_sliding_strings.string [hunk 3]
```diff
@@ -1,41 +1,202 @@
...
+This really takes me back to our\nfirst battle. Do you remember it,\ntoo?\p 
 
-#org @sText_MaximaLastSwitchIn
-The power of Mega Evolution[.]\nCan you feel it within your reach?\p 
+#org @sText_ChampionRematch_LastSwitchIn
+It's quite refreshing to see how\nstrong a bond you share with your\nPokémon.\p 
 
-#endif
+#org @sText_ChampionRematch_LowHP
+I may not be the champion any\nlonger, but it wouldn't be very\nsporting for me to give up now!\p
```
Comment (CrystalAQuick) [inline strings/trainer_sliding_strings.string:202]: " but it wouldn't be very sporting for me to give up now!" sounds off to me

File: include/new/trainer_sliding_data.h [hunk 4]
```diff
@@ -14,14 +14,99 @@ extern const u8 BattleScript_TrainerSlideMsgEnd2[];
...
-extern const u8 sText_GalavanLastSwitchIn[];
-extern const u8 sText_TessyLastSwitchIn[];
-extern const u8 sText_Zeph1LowHP[];
-extern const u8 sText_Zeph2LowHP[];
-extern const u8 sText_MaximaLastSwitchIn[];
+// Rival battle 1
+extern const u8 sText_Rival1_LastSwitchIn[];
+extern const u8 sText_Rival1_LowHP[];
+
+// Rival battle n
```
Comment (CrystalAQuick) [inline include/new/trainer_sliding_data.h:21]: why is this comment here? 
// Rival battle n

File: include/new/trainer_sliding_data.h [hunk 5]
```diff
@@ -14,14 +14,99 @@ extern const u8 BattleScript_TrainerSlideMsgEnd2[];
...
+extern const u8 sText_Leader7_LastSwitchIn[];
+extern const u8 sText_Leader7_LowHP[];
+
+// Gym Leader 8 (Dennis & Dee)
+extern const u8 sText_Leader8_FirstMonDown[];
+extern const u8 sText_Leader8_LastSwitchIn[];
+extern const u8 sText_Leader8_LowHP[];
+
+// Team Pluto Admin Irene
+extern const u8 sText_Irene_FirstMonDown[];
```
Comment (CrystalAQuick) [inline include/new/trainer_sliding_data.h:64]: some areas you are doing things like
extern const u8 sText_Leader1_FirstMonDown[];  
(note how the name of the leader is not included)

when you are refering to a particular person in other places you are doing 
extern const u8 sText_Irene_FirstMonDown[]; 

my suggestion is to be consistent with the naming.
extern const u8 sText_PlutoAdmin_Irene_FirstMonDown[]; 
extern const u8 sText_GymLeader_Terrence_FirstMonDown[];  
..
or 
extern const u8 sText_Leader1_FirstMonDown[];  
extern const u8 sText_PlutoAdmin1_FirstMonDown[]; 
extern const u8 sText_PlutoLeader_FirstMonDown[]; 

so tdlr: drop the name all together or include it in places you have not.  this change will have to get done in many places

**Review by CrystalAQuick — CHANGES_REQUESTED** [review]

File: strings/trainer_sliding_strings.string (hunk 3 above)
Comment (assassinsgreed) [reply strings/trainer_sliding_strings.string]: Changed it up a bit.  LMK what you think

File: include/new/trainer_sliding_data.h (hunk 4 above)
Comment (assassinsgreed) [reply include/new/trainer_sliding_data.h:21]: It's a placeholder for now, until we know how many rival battles we'll have

File: include/new/trainer_sliding_data.h (hunk 5 above)
Comment (assassinsgreed) [reply include/new/trainer_sliding_data.h:64]: I'm partial to keeping it this way, at least for now, to slim up on characters.  IMO the trainer name (or # for something more regular like gym leader order) is clear enough to read.  LMK what you think.

**Comment by CrystalAQuick** [comment]
Looks good

**Review by CrystalAQuick — APPROVED** [review]

**Pull request merged**